core/src/crds.rs

```diff
@@ -172,7 +173,7 @@ impl Crds {
                         self.nodes.insert(entry_index);
                     }
                     CrdsData::Vote(_, _) => {
-                        self.votes.insert(entry_index);
+                        self.votes.insert(value.ordinal, entry_index);
```

Comment:
For clarification, so even though the `IndexSet` tracks the order of the insertions, there's no way to look up all keys inserted *after* a specific key `k`, which is where the `BtreeMap` has the advantage?

Reply:
Current code only uses `IndexSet` because of faster sequential iterations (due to cache locality). It does not actually preserve insertion order. For example the `swap_remove` in these lines:
https://github.com/solana-labs/solana/blob/8aec50a27/core/src/crds.rs#L410
https://github.com/solana-labs/solana/blob/8aec50a27/core/src/crds.rs#L444
and, also in the case a value is updated, `self.votes` does not change:
https://github.com/solana-labs/solana/blob/8aec50a27/core/src/crds.rs#L188-L203

So, ultimately current index does not maintain insertion orders.